### ISSUE TITLE
[Bug]: import matplotlib.backends.backend_qt5agg error for pyside6

### ISSUE THREAD

**Issue by koutoftimer** [matplotlib/matplotlib#24212]
### Bug summary

https://github.com/matplotlib/matplotlib/blob/318b2348f7127efc3e28aadebf762aae32b6343c/lib/matplotlib/backends/qt_compat.py#L144

Have to check for "PySide6" too.

### Code for reproduction

```python
$ QT_API=pyside6 python -c "import matplotlib.backends.backend_qt5agg"
```

### Actual outcome

```
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/tmp/tmp.atoQSuHf68/venv/lib/python3.10/site-packages/matplotlib/backends/backend_qt5agg.py", line 7, in <module>
    from .backend_qtagg import (    # noqa: F401, E402 # pylint: disable=W0611
  File "/tmp/tmp.atoQSuHf68/venv/lib/python3.10/site-packages/matplotlib/backends/backend_qtagg.py", line 12, in <module>
    from .backend_qt import QtCore, QtGui, _BackendQT, FigureCanvasQT
  File "/tmp/tmp.atoQSuHf68/venv/lib/python3.10/site-packages/matplotlib/backends/backend_qt.py", line 72, in <module>
    _MODIFIER_KEYS = [
  File "/tmp/tmp.atoQSuHf68/venv/lib/python3.10/site-packages/matplotlib/backends/backend_qt.py", line 73, in <listcomp>
    (_to_int(getattr(_enum("QtCore.Qt.KeyboardModifier"), mod)),
TypeError: int() argument must be a string, a bytes-like object or a real number, not 'KeyboardModifier'
```

### Expected outcome

no errors

### Additional information

PySide6==6.4.0

### Operating system

GNU\Gentoo Linux

### Matplotlib Version

3.6.1

### Matplotlib Backend

same exception

### Python version

3.10.6

### Jupyter version

not installed

### Installation

pip

**Comment by koutoftimer** [comment]
I suppose it is already fixed in the main branch.

https://github.com/matplotlib/matplotlib/blob/80b15fb17be6455680a4830d0644731f8ee6bb8b/lib/matplotlib/backends/qt_compat.py#L71-L115

**Comment by koutoftimer** [comment]
Yes it is fixed already. Can't reproduce the issue with main branch.

``` 
$ pip freeze
matplotlib @ git+https://github.com/matplotlib/matplotlib.git@80b15fb17be6455680a4830d0644731f8ee6bb8b
PySide6==6.4.0
```

For me works fine.

**Comment by oscargus** [comment]
Was fixed in #24158
